src/main/java/github/scarsz/discordsrv/api/ApiManager.java

```diff
@@ -247,6 +247,12 @@ public void updateSlashCommands() {
      * @param provider the command data provider
      */
     public void addSlashCommandProvider(@NonNull SlashCommandProvider provider) {
+        // Prevent double registration, if an instance is already a plugin instance
+        for (Plugin plugin : Bukkit.getPluginManager().getPlugins()) {
+            if (provider == plugin) {
+                return;
+            }
+        }
         this.slashCommandProviders.add(provider);
```

Comment:
`slashCommandProviders` is a set that doesn't accept duplicates, this check is unnecessary